repl/src/main/scala/com/cloudera/livy/repl/ReplDriver.scala

```diff
@@ -70,22 +73,16 @@ class ReplDriver(conf: SparkConf, livyConf: RSCConf)
   /**
    * Return statement results. Results are sorted by statement id.
    */
-  def handle(ctx: ChannelHandlerContext, msg: BaseProtocol.GetReplJobResults): ReplJobResults =
-    session.synchronized {
-      val stmts = if (msg.allResults) {
-        session.statements.values.toArray
-      } else {
-        assert(msg.from != null)
-        assert(msg.size != null)
-        val until = msg.from + msg.size
-        session.statements.filterKeys(id => id >= msg.from && id < until).values.toArray
-      }
-      val state = session.state.toString
-      new ReplJobResults(stmts.sortBy(_.id), state)
+  def handle(ctx: ChannelHandlerContext, msg: BaseProtocol.GetReplJobResults): ReplJobResults = {
+    val statements = if (msg.allResults) {
+      session.statements.values.toArray
+    } else {
+      assert(msg.from != null)
+      assert(msg.size != null)
+      val until = msg.from + msg.size
+      session.statements.filterKeys(id => id >= msg.from && id < until).values.toArray
```

Comment:
From my understanding in most of the cases querying one statement is enough, compared to the previous code if we only want to get one statement, here we still requires to scan the map, this might be a regression IIUC.

 Can we separate the scenario for querying one statement, using another protocol to handle this issue?